gapis/server/grpc.go

```diff
@@ -365,3 +366,72 @@ func (s *grpcServer) ClientEvent(ctx xctx.Context, req *service.ClientEventReque
 	}
 	return &service.ClientEventResponse{}, nil
 }
+
+func (s *grpcServer) TraceTargetTreeNode(ctx xctx.Context, req *service.TraceTargetTreeRequest) (*service.TraceTargetTreeResponse, error) {
+	defer s.inRPC()()
+	res, err := s.handler.TraceTargetTreeNode(s.bindCtx(ctx), req)
+	if err := service.NewError(err); err != nil {
+		return &service.TraceTargetTreeResponse{Val: &service.TraceTargetTreeResponse_Error{Error: err}}, nil
+	}
+	return &service.TraceTargetTreeResponse{Val: &service.TraceTargetTreeResponse_Node{Node: res}}, nil
+}
+
+func (s *grpcServer) FindTraceTarget(ctx xctx.Context, req *service.FindTraceTargetRequest) (*service.TraceTargetTreeResponse, error) {
+	defer s.inRPC()()
+	res, err := s.handler.FindTraceTarget(s.bindCtx(ctx), req)
+	if err := service.NewError(err); err != nil {
+		return &service.TraceTargetTreeResponse{Val: &service.TraceTargetTreeResponse_Error{Error: err}}, nil
+	}
+	return &service.TraceTargetTreeResponse{Val: &service.TraceTargetTreeResponse_Node{Node: res}}, nil
+}
+
+func (s *grpcServer) Trace(conn service.Gapid_TraceServer) error {
+	ctx := s.bindCtx(conn.Context())
+	t, err := s.handler.Trace(ctx)
+	if err != nil {
+		return err
+	}
+	sig, f := task.NewSignal()
```

Comment:
Is there any need for the signal + goroutine? Looks completely synchronous at first glance.

Reply:
I think you are right!